aziotd/src/main.rs

```diff
@@ -186,58 +227,154 @@ where
 }
 
 fn merge_toml(base: &mut toml::Value, patch: toml::Value) {
-    // Similar to JSON patch, except that maps are called tables, and
-    // there is no equivalent of null that can be used to remove keys from an object.
+    // Similar to JSON patch, except that:
+    //
+    // - Maps are called tables.
+    // - There is no equivalent of null that can be used to remove keys from an object.
+    // - Arrays are merged via concatenating the patch to the base, rather than replacing the base with the patch.
 
-    if let toml::Value::Table(base) = base {
+    if let toml::Value::Table(base) = &mut *base {
```

Comment:
Is the `&mut *base` here to clarify that `base` has type `&mut`, or is there some other reason I'm missing?

If it's the former, then it probably best to just leave it as `base`, and keep things terse.

Reply:
`&mut` is not `Copy` so it moves by default unless it's reborrowed.

Reply:
Would `if let toml::Value::Table(ref mut base) = base` do the right thing in this case?

Reply:
Actually, I must've shuffled some of the code around after that because I don't see that error any more, so I'll just take it out.